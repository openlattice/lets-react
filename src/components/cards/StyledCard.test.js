import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import StyledCard from './StyledCard';

describe('StyledCard', () => {

  let props;
  let mountedStyledCard;

  const styledCard = () => {
    if (!mountedStyledCard) {
      mountedStyledCard = mount(
        <StyledCard {...props} />
      );
    }
    return mountedStyledCard;
  };

  beforeEach(() => {
    props = {

    };
    mountedStyledCard = undefined;
  });

  it('renders without throwing an error', () => {
    const tree = renderer.create(<StyledCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = styledCard().find('div');
    expect(divs).toHaveLength(1);
  });

});
