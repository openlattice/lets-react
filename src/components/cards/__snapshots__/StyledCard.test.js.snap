// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCard renders without throwing an error 1`] = `
.c0 {
  background-color: #fefefe;
  border: 1px solid #c5d5e5;
  border-radius: 4px;
  box-shadow: 0 2px 8px -2px rgba(17,51,85,0.15);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  font-weight: normal;
  padding: 30px;
  margin: 0 20px;
}

.c0 h1 {
  font-size: 22px;
  font-weight: 600;
  margin: 10px 0;
  padding: 0;
}

.c0 h2 {
  font-size: 16px;
  font-weight: 600;
  margin: 18px 0 10px 0;
  padding: 0;
}

.c0 h3 {
  font-size: 14px;
  font-weight: 600;
  margin: 16px 0 8px 0;
  padding: 0;
}

.c0 p {
  margin: 0;
  padding: 0;
}

.c0 section {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 10px;
  margin-top: 10px;
}

<div
  className="c0"
/>
`;
